### ISSUE TITLE
NGrams doesnt support words with hyphen and slash in English

### ISSUE THREAD

**Issue by sam68740** [NaturalNode/natural#656]
There are a few words in English that contain hyphen or slash

Example:

- image-based
- text-based
- links/CTA

It would be great if Natural could manage these cases.

```
let text = "links text-based opposed image-based links/CTA’s"
var NGrams = natural.NGrams;
const T = natural.AggressiveTokenizer;
const tokenizer = new T();
NGrams.setTokenizer(tokenizer);
console.log(NGrams.ngrams(text, 1));
```

Output:  [["links"], ["text"], ["based"], ["opposed"], ["image"], ["based"], ["links"], ["CTA"], ["s"]]

**Comment by Hugo-ter-Doest** [comment]
The root cause of this behaviour is the tokenizer. Will look into adapting the tokenizer to support these characters.

**Comment by sam68740** [comment]
There is also other issues in other language.
For instance the Eszett symbol ß is also not supported.
If you can also check, it will be nice.

Thanks

**Comment by Hugo-ter-Doest** [comment]
Solved in #706
